python/afdko/buildcff2vf.py

```diff
@@ -1,998 +1,486 @@
+        return new_coords
+
+    def check_and_fix_flat_curve(self, cmd, point_type, pt_coords):
+        success = False
+        if (point_type == 'rlineto') and (cmd[0] == 'rrcurveto'):
+            is_default = False  # the line is in the master font we are adding
+            pt_coords = self.make_curve_coords(pt_coords, is_default)
+            success = True
+        elif (point_type == 'rrcurveto') and (cmd[0] == 'rlineto'):
+            is_default = True  # the line is in the default font commands
+            expanded_coords = self.make_curve_coords(cmd[1], is_default)
+            cmd[1] = expanded_coords
+            cmd[0] = point_type
+            success = True
+        return success, pt_coords
+
+    def check_and_fix_closepath(self, cmd, point_type, pt_coords):
+        """ Some workflows drop a lineto which closes a path.
+        Also, if the last segment is a curve in one master,
+        and a flat curve in another, the flat curve can get
+        converted to a closing lineto, and then dropped.
+        Test if:
+        1) one master op is a moveto,
+        2) the previous op for this master does not close the path
+        3) in the other master the current op is not a moveto
+        4) the current op in the otehr master closes the current path
+
+        If the default font is missing the closing lineto, insert it,
+        then proceed with merging the current op and pt_coords.
+
+        If the current region is missing the closing lineto
+        and therefore the current op is a moveto,
+        then add closing coordinates to self._commands,
+        and increment self.pt_index.
+
+        Note that if this may insert a point in the default font list,
+        so after using it, 'cmd' needs to be reset.
+
+        return True if we can fix this issue.
+        """
+        if point_type == 'rmoveto':
+            # If this is the case, we know that cmd[0] != 'rmoveto'
+
+            # The previous op must not close the path for this region font.
+            prev_moveto_coords = self._commands[self.prev_move_idx][1][-1]
+            prv_coords = self._commands[self.pt_index - 1][1][-1]
+            if prev_moveto_coords == prv_coords[-2:]:
+                return False
+
+            # The current op must close the path for the default font.
+            prev_moveto_coords2 = self._commands[self.prev_move_idx][1][0]
+            prv_coords = self._commands[self.pt_index][1][0]
+            if prev_moveto_coords2 != prv_coords[-2:]:
+                return False
+
+            # Add the closing line coords for this region
+            # so self._commands, then increment self.pt_index
+            # so that the current region op will get merged
+            # with the next default font moveto.
+            if cmd[0] == 'rrcurveto':
+                new_coords = self.make_curve_coords(prev_moveto_coords, False)
+            cmd[1].append(new_coords)
```

Comment:
This line should be indented to the same level as the previous line, as `new_coords` is only assigned within the `if` block.

Reply:
Right.

Reply:
OK, I see that this is fixed now, so resolving this conversation.